packages/ssr/src/analytics-providers/google-analytics.ts

```diff
@@ -0,0 +1,67 @@
+import type { AnalyticsEvent } from '../analytics-events';
+import axios, { AxiosResponse } from 'axios';
+
+const ga4Axios = axios.create({
+  baseURL: 'https://www.google-analytics.com',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+const url = `/mp/collect?api_secret=${process.env.GA_API_TOKEN}`
+  + `&measurement_id=${process.env.GA_MEASUREMENT_ID}`;
+
+const sendEvent = (
+  analyticsEvents: AnalyticsEvent[],
+  clientId: string,
+  sessionId: string,
+  utm?: {
+    source?: string,
+    medium?: string,
+    campaign?: string,
+    term?: string,
+    content?: string
+  },
+) => {
+  if (process.env.GA_API_TOKEN && process.env.GA_MEASUREMENT_ID) {
+    const requests: Promise<AxiosResponse<any, any>>[] = [];
+    let events: AnalyticsEvent[] = [];
+
+    analyticsEvents.forEach((event, index) => {
+      if (event.params) {
+        Object.assign(event.params, { session_id: sessionId });
+      }
+
+      const body = event;
+      if (body.name === 'page_view') {
+        body.name = 'campaign_details';
+        body.params = {
+          source: utm?.source || event.params?.page_title,
```

Comment:
```suggestion
          source: utm?.source,
```

UTM source tem nada a ver com page title 😄